### PR TITLE
Migrate acos operation from ngraph backend test to template plugin reference test

### PR DESCRIPTION
**Pull request by wilson-seok**
### Details:
 - Create template plugin reference test for target operation Acos-1
 - Check CPU/GPU SLT parameter
 - Create visitor API test for target operation Acos-1
 - Delete ngraph backend test for target operation Acos-1
 - Remove boolean type from acos_evaluate

### Tickets:
 - 63940

### CODE REVIEW

File: docs/template_plugin/tests/functional/op_reference/acos.cpp [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+TEST_P(ReferenceAcosLayerTest, AcosWithHardcodedRefs) {
+    Exec();
+}
+
+}  // namespace
+
+INSTANTIATE_TEST_SUITE_P(
+    smoke_Acos_With_Hardcoded_Refs, ReferenceAcosLayerTest,
+    ::testing::Values(Builder {}
+                          .input({{11}, element::f16, std::vector<ngraph::float16> {-1.f, -0.75f, -0.5f, -0.25f, -0.125f,
```
Comment (ilyachur) [inline docs/template_plugin/tests/functional/op_reference/acos.cpp:63]: Could you cover all supported data types?

Comment (wilson-seok) [reply docs/template_plugin/tests/functional/op_reference/acos.cpp:63]: input vector for int will have only {-1, 0, 1}, so only 3 elements are ok for test?

Comment (wilson-seok) [reply docs/template_plugin/tests/functional/op_reference/acos.cpp:63]: Add cases for int32, int64, uint32, uint64.

Comment (ilyachur) [reply docs/template_plugin/tests/functional/op_reference/acos.cpp:63]: As I see acos evaluate supports next data types:
```
    case ngraph::element::i32:
    case ngraph::element::i64:
    case ngraph::element::u32:
    case ngraph::element::u64:
    case ngraph::element::f16:
    case ngraph::element::f32:
    case ngraph::element::boolean:
```
Can you cover all supported data types?

Comment (wilson-seok) [reply docs/template_plugin/tests/functional/op_reference/acos.cpp:63]: I can see support data type includes boolean type in ov::op::v0::Acos::has_evaluate(). But couldn't find any test case of boolean type in reference test cases of other trigonometric function operations. Is it enough to use {false, true} input as {0, 1}?

Comment (wilson-seok) [reply docs/template_plugin/tests/functional/op_reference/acos.cpp:63]: When I tried element::boolean in the op reference test, error happened in validate_and_infer_elementwise_arithmetic(). It doesn't allow the element to have boolean when is_dynamic()=false.

https://github.com/wilson-seok/openvino/blob/f347f83b0a02adb47a1ffdd38c81ed2b393b8773/ngraph/core/src/op/util/unary_elementwise_arithmetic.cpp#L16

Comment (wilson-seok) [reply docs/template_plugin/tests/functional/op_reference/acos.cpp:63]: @ilyachur Could you please confirm that we can skip boolean type test in the reference test?

Comment (ilyachur) [reply docs/template_plugin/tests/functional/op_reference/acos.cpp:63]: I think that we should remove boolean from the evaluate method.

Comment (wilson-seok) [reply docs/template_plugin/tests/functional/op_reference/acos.cpp:63]: Ok, I removed it from evaluate_acos().

**Review by ilyachur — APPROVED** [review]

**Comment by wilson-seok** [comment]
@ilyachur Could you please merge this PR if you have the permission?

**Pull request merged**